crates/env/src/lib.rs

```diff
@@ -114,9 +114,9 @@ macro_rules! debug_print {
 /// with a newline appended.
 #[macro_export]
 macro_rules! debug_println {
-    () => ($crate::print!("\n"));
+    () => ($crate::debug_print!("\n"));
     ($($arg:tt)*) => ({
-        $crate::debug_print!($($arg)*);
-        $crate::debug_print!("\n");
+        let msg = $crate::ink_prelude::format!($($arg)*);
```

Comment:
does this compile? doesn't `$crate` stand for the crate? Don't we rather need `::ink_prelude::format!(..)`? Maybe I am missing some Rust rules here.

Reply:
Using `::ink_prelude` depends on the contract importing that crate, which is not guaranteed. `$crate::ink_prelude` resolves to the reexported `::ink::env::ink_prelude`.